src/kudos/controllers/kudos.controller.ts

```diff
@@ -0,0 +1,163 @@
+import {KudosGivenDto} from "../dto/kudos-given.dto";
+import {PostSlackDto} from "../dto/post-slack.dto";
+import {UserService} from "../services/user.service";
+import {DialogPostSlackDto, PayloadClass} from "../dto/dialog-post-slack.dto";
+import {SlackService} from "../services/slack.service";
+import {SingleKudosSlackDto} from "../dto/single-kudos-slack.dto";
+import {SlackTokenGuard} from "../guards/slackToken.guard";
+import {KudosDto} from "../dto/kudos.dto";
+import {ApiBearerAuth, ApiForbiddenResponse, ApiResponse, ApiUseTags} from '@nestjs/swagger';
+import {AvatarDto} from "../dto/avatar.dto";
+
+@Controller('kudos')
+@ApiUseTags('kudos')
+export class KudosController {
+
+    constructor(private kudosService: KudosService,
+                private userService: UserService,
+                private slackService: SlackService) {
+    }
+
+    @Get()
+    @ApiResponse({status: 200, description: 'Get all Kudos', type: KudosDto, isArray: true})
+    async getKudos(): Promise<KudosDto[]> {
+        return await this.kudosService.getAllWithAvatars();
+    }
+
+    @Get('rankings')
+    @ApiResponse({status: 200, description: 'Get overall rankings', type: KudosRankingDto, isArray: true})
+    async getRankings(): Promise<KudosRankingDto[]> {
+        const kudos = await this.kudosService.getRankings();
+        const users = await this.userService.findByUsersName(kudos.map(el => el.name))
+        return kudos.map(({name, totalPoints}) => ({
+                user: {
+                    name: name,
+                    avatar: this.userService.mapAvatarToAvatarDto(users.find(u => u.name == name))
+                },
+                totalPoints: Number(totalPoints)
+            }
+        ));
+    }
+
+    @Get('from')
+    @ApiResponse({
+        status: 200,
+        description: 'Kudos `from` overall statistics for months/years',
+        type: KudosFromDto,
+        isArray: true
+    })
+    async kudosFromUsers(): Promise<KudosFromDto[]> {
+        const kudos = await this.kudosService.getFrom();
+        const users = await this.userService.findByUsersName(kudos.map(el => el.name))
+        return kudos.map(({quantity, name, month, year}) =>
+            ({
+                quantity: Number(quantity), year, from: {
+                    name: name,
+                    avatar: this.userService.mapAvatarToAvatarDto(users.find(u => u.name == name))
+                },
+                month
+            }));
+    }
+
+    @Get('given')
+    @ApiResponse({
+        status: 200,
+        description: 'Kudos `given` overall statistics for months/years',
+        type: KudosGivenDto,
+        isArray: true
+    })
+    async kudosGivenToUsers(): Promise<KudosGivenDto[]> {
+        const kudos = await this.kudosService.getGiven();
+        const users = await this.userService.findByUsersName(kudos.map(el => el.name))
+        return kudos.map(({quantity, name, month, year}) => ({
+                quantity: Number(quantity), year, givenTo: {
```

Comment:
Często zamieniasz `quantity` na liczbę. Warto byłoby przenieść to zachowanie do `KudosService`, zamiast wymagać tej transformacji od wszystkich klientów.